sdk/textanalytics/Azure.AI.TextAnalytics.Protocol/README.md

```diff
@@ -12,12 +12,307 @@ At this time, if you are looking to interact with the Azure Cognitive Services T
+
+```csharp
+var objectDoc = new JsonData(new { key1 = "one", key2 = "two" }); // represents the document: { "key1": "one", "key2": "two" }
+Console.WriteLine(objectDoc.Get("key1")));               // prints "one"
+Console.WriteLine(objectDoc.Get("missingKey") == null)); // prints "true"
+Console.WriteLine(objectDoc["missingKey"]);              // throws InvalidOperationException.
+```
+
+Arrays are similar to Objects, you can use the indexer to fetch a specific element from the array:
+
+```csharp
+var arrayDoc = new JsonData(new[] { "Hello", "JsonData" }); // represents the document: [ "Hello", "JsonData" ]
+Console.WriteLine(arrayDoc[0]);   // prints "Hello"
+Console.WriteLine(arrayDoc[1]);   // prints "JsonData"
+```
+
+You can also use the `Items` property to get an `IEnumerable<JsonData>`, which you can use to enumerate the values:
+
+```csharp
+var arrayDoc = new JsonData(new[] { "Hello", "JsonData" }); // represents the document: [ "Hello", "JsonData" ]
+foreach (JsonData item in arrayDoc.Items)
+{
+    Console.WriteLine(item);
+}
+```
+
+Combined, this provides a nice way to access values.  Consider the JsonData that represents the document:
+
+```json
+{
+  "students": [
+      {
+          "name": "Matt",
+          "address": [ "1 Microsoft Way", "Building 18", "Redmond, WA, 98034" ]
+      },
+      {
+          "name": "Bill",
+          "address": [ "1 Microsoft Way", "Building 34", "Redmond, WA, 98034" ]
+      }
+  ]
+}
+```
+
+You can access the inner values with ease:
+
+```csharp
+JsonData doc = JsonData.FromString(/* a string representing the above document */);
+Console.WriteLine(doc["students"][0]["name"]); // prints "Matt"
+Console.WriteLine(doc["students"][1]["address"][1]); // prints "Building 34"
+```
+
+##### Using `dynamic`
+
+You may also use the `dynamic` keyword with `JsonData`, and it behaves as you might expect:
+
+```csharp
+dynamic doc = JsonData.FromString(/* a string representing the above document */);
+Console.WriteLine(doc.students[0].name); // prints "Matt"
+Console.WriteLine(doc.students[1].address[1]); // prints "Building 34"
+```
+
+And you can cast back to a CLR type as you might expect:
+
+```csharp
+dynamic doc = JsonData.FromString(/* a string representing the above document */);
+string name = (string)doc.students[0].name; // name is set to the string "Matt"
+string address = (string) doc.students[1].address[1]; // address is set to the string "Building 34"
+```
+
+##### Conversions to and from POCOs
+
+As we saw earlier, `JsonData` can be constructed from an arbitrary CLR object. When this happens, the object is serialized using `System.Text.Json`. You can also convert a JsonData *to* a CLR object:
+
+```csharp
+class Student
+{
+    string name { get; }
+    string address { get; }
+}
+
+dynamic doc = JsonData.FromString(/* a string representing the above document */);
+Student[] students = doc["students"].To<Student[]>();
```

Comment:
this is a bit confusing given doc is dynamic. BTW, for dynamic references, we should allow 
Students[] students = (Student[])doc.students;

does that work? If not, we need to implement the dynamic cast.

Reply:
It does not work yet but I can implement it.

Reply:
> This is a bit confusing given doc is dynamic.

Yup, sorry. that was a copy/paste bug, I updated the sample so that it's explicitly typed as `JsonData` here, so it will be less confusing.

Reply:
Fixed the copy/paste error.  The dynamic cast support to model types will come in the next PR.